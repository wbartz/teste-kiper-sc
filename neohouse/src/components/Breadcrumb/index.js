import PropTypes from 'prop-types';
import React, { useEffect, useState, useCallback } from 'react';
import { withRouter } from 'react-router-dom';
import './index.scss';

const Breadcrumb = ({ history, location }) => {
  const [pages, setPages] = useState('');

  const getPath = (path) =>
    (path.slice(0, 1).toUpperCase() + path.slice(1)).replace('-', ' ');

  const getPages = useCallback((url) => {
    const routes = url.slice(1).split('/');
    const paths = [];

    if (routes.length >= 1) {
      paths.push({
        label: 'Dashboard',
        link: '/dashboard',
        active: routes.length === 1,
      });
    }

    if (routes.length >= 2) {
      paths.push({
        label: `${getPath(routes[0])} - ${getPath(routes[1])}`,
        link: `/${routes[0]}/${routes[1]}`,
        active: routes.length === 2,
      });
    }

    if (routes.length === 4) {
      paths.push({
        label: `${getPath(routes[2])} - ${getPath(routes[3])}`,
        link: `/${routes[3]}/${routes[2]}`,
        active: true,
      });
    }
    setPages(paths);
  }, []);

  useEffect(() => {
    getPages(location.pathname);
  }, [location, getPages]);

  return (
    <div>
      <div className="breadcrumb">
        {pages
          ? pages.map((page) => (
              <span key={page.label} className={page.active ? 'active' : ''}>
                {!page.active ? (
                  <i onClick={() => history.push(`${page.link}`)}>
                    {page.label}
                  </i>
                ) : (
                  page.label
                )}
              </span>
            ))
          : null}
      </div>
    </div>
  );
};

Breadcrumb.propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
};

export default withRouter(Breadcrumb);
