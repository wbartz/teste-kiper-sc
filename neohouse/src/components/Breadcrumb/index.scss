@import '../../styles/variables';

.breadcrumb {
  margin-top: 30px;
  margin-bottom: 30px;
  display: flex;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  width: 100%;
  padding: 10px;
  border-color: $color-lighter-gray;

  span {
    font-style: italic;
    color: $color-gray;
    margin-right: 10px;

    a {
      cursor: pointer;
      color: $color-secondary;
      font-size: $h5-font-size;
    }

    a::after {
      font-size: $h4-font-size;
      font-style: italic;
      font-weight: bold;
      content: " | ";
    }
  }

  span.active {
    color: $color-alternative;
    
  }
}