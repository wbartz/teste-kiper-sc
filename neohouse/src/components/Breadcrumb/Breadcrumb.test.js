import { shallow } from 'enzyme';
import React from 'react';
import Breadcrumb from '.';

describe('Breadcrumb', () => {
  const props = {
    onRedirect: jest.fn(),
    pages: [
      {
        label: 'teste',
        link: '/test',
      },
    ],
    active: 'teste 2',
  };

  it('render without crashing', () => {
    expect(shallow(<Breadcrumb {...props} />)).toMatchSnapshot();
  });
});
