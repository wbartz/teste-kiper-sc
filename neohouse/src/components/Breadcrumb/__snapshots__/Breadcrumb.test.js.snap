// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb render without crashing 1`] = `ShallowWrapper {}`;
